zerver/actions/message_send.py

```diff
@@ -1528,6 +1529,37 @@ def validate_stream_id_with_pm_notification(
     return stream
 
 
+def check_can_send_private_message(
+    realm: Realm, sender: UserProfile, user_profiles: Sequence[UserProfile], recipient: Recipient
+) -> None:  # nocoverage
+    if realm.name == "Realm Test" or sender.is_bot:
+        return
```

Comment:
This function needs a complete test suite with 100% coverage and to be cleaned up to not hardcode particular values. In general you should write tests before asking for review on a PR -- I know it might seem like a chore if the design changes -- but helps catch bugs, and often it's actually easiest to start by reading the tests to see what corner cases the author has thought about.

Reply:
Added tests in ``` test_message_send.py``` for these settings in the recent changes.